From bdb372144fb91710be08277c832a32e4806a9e8c Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Thu, 25 Oct 2012 14:54:39 +0800
Subject: [PATCH 086/281] ENGR00222067 build: change to use fs_mgr to mount

1.This patch change to use fs_mgr to manage filesystem. this is new feature in
JB. If still use old style,"nonencrypted" will not be triggered after mounting
system. It will cause class late_start svc not launched rightly.
2.Clear warning message caused by svc bootanim.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 imx6/etc/fstab.freescale | 11 +++++++++++
 imx6/etc/init.rc         |  2 +-
 imx6/imx6.mk             |  1 +
 imx6/sabresd/init.rc     |  6 +-----
 4 files changed, 14 insertions(+), 6 deletions(-)
 create mode 100644 imx6/etc/fstab.freescale

diff --git a/imx6/etc/fstab.freescale b/imx6/etc/fstab.freescale
new file mode 100644
index 0000000..6b70843
--- /dev/null
+++ b/imx6/etc/fstab.freescale
@@ -0,0 +1,11 @@
+# Android fstab file.
+#<src>                                                  <mnt_point>         <type>    <mnt_flags>                                                                         <fs_mgr_flags>
+# The filesystem that contains the filesystem checker binary (typically /system) cannot
+# specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
+
+
+/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
+/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
+/dev/block/mmcblk0p7    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait
+/dev/block/mmcblk0p8    /device  ext4    ro,nosuid,nodev				                                  wait
+
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 6f78217..39c62e3 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -486,7 +486,7 @@ service media /system/bin/mediaserver
 
 service bootanim /system/bin/bootanimation
     class main
-    user graphics
+    user system
     group graphics
     disabled
     oneshot
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index e7059e7..43e36db 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -260,6 +260,7 @@ PRODUCT_COPY_FILES +=	\
 	device/fsl/common/input/eGalax_Touch_Screen.idc:system/usr/idc/eGalax_Touch_Screen.idc \
 	device/fsl/common/input/eGalax_Touch_Screen.idc:system/usr/idc/HannStar_P1003_Touchscreen.idc \
 	device/fsl/imx6/etc/init.rc:root/init.rc \
+	device/fsl/imx6/etc/fstab.freescale:root/fstab.freescale \
 	device/fsl/imx6/etc/apns-conf.xml:system/etc/apns-conf.xml \
 	device/fsl/imx6/etc/init.usb.rc:root/init.freescale.usb.rc \
 	device/fsl/imx6/etc/ueventd.freescale.rc:root/ueventd.freescale.rc \
diff --git a/imx6/sabresd/init.rc b/imx6/sabresd/init.rc
index 03bd13c..d686c73 100755
--- a/imx6/sabresd/init.rc
+++ b/imx6/sabresd/init.rc
@@ -122,8 +122,4 @@ service iprenew_wlan0 /system/bin/dhcpcd -n
 
 on fs
 # mount ext4 partitions
-    mount ext4 /dev/block/mmcblk0p5 /system
-    mount ext4 /dev/block/mmcblk0p5 /system ro remount
-    mount ext4 /dev/block/mmcblk0p7 /data nosuid nodev nodiratime noatime noauto_da_alloc,errors=panic
-    mount ext4 /dev/block/mmcblk0p6 /cache nosuid nodev
-    mount ext4 /dev/block/mmcblk0p8 /device ro nosuid nodev
+    mount_all /fstab.freescale
-- 
1.8.0

